app/models/manageiq/providers/redhat/infra_manager/api_integration.rb

```diff
@@ -273,9 +273,17 @@ def raw_connect_v4(server, port, path, username, password, service, scheme = 'ht
       # Get the timeout from the configuration:
       timeout, = ems_timeouts(:ems_redhat, service)
 
+      # Build the URL:
+      url = URI::Generic.build(
+        :scheme => scheme,
+        :host   => server,
+        :port   => port,
+        :path   => path
+      )
+
       # Create the connection:
       OvirtSDK4::Connection.new(
-        :url      => "#{scheme}://#{server}:#{port}#{path}",
+        :url      => url.to_s,
```

Comment:
The OvirtSDK4::Connection.new constructor may call or not URI.parse, that is an implementation detail. Currently the contract is that it receives an string.